### PR TITLE
Cu 27ctvhk public userid mina lee

### PR DESCRIPTION
**Pull request by JM066**
# Description
User page added - display user's info and projects that redirect to the project detail page

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by ivanms1** [comment]
Task linked: [CU-27ctvhk public /user/[id]](https://app.clickup.com/t/27ctvhk)

File: apps/web/pages/user/[id].tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import User from '@/pages/User';
+
+export default User;
+
+export async function getServerSideProps(context) {
+  const { params } = context;
+  const id = params.id;
```
Comment (ivanms1) [inline apps/web/pages/user/[id].tsx:7]: you should call the query here

File: apps/web/src/pages/User/index.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            alt={user?.name}
+            height={200}
+            width={200}
+          />
+        )}
+        <StyledTitle>{user?.name}</StyledTitle>
+      </StyledUserContainer>
+      <StyledProjectContainer>
+        {user?.projects.map((project) => (
+          <StyledLink key={project.id}>
```
Comment (ivanms1) [inline apps/web/src/pages/User/index.tsx:42]: you can reuse the `ProjectCard` component

File: apps/web/src/pages/User/index.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+                  height={300}
+                  width={400}
+                />
+                <div>{project.title}</div>
+              </StyledCard>
+            </Link>
+          </StyledLink>
+        ))}
+      </StyledProjectContainer>
+    </StyledUser>
```
Comment (ivanms1) [inline apps/web/src/pages/User/index.tsx:57]: Please add SEO

**Review by ivanms1 — CHANGES_REQUESTED** [review]
Good job!
Please check my comments

File: apps/web/pages/user/[id].tsx (hunk 1 above)
Comment (JM066) [reply apps/web/pages/user/[id].tsx:7]: Thanks! It is done!

File: apps/web/src/pages/User/index.tsx (hunk 2 above)
Comment (JM066) [reply apps/web/src/pages/User/index.tsx:42]: Replaced with ProjectCard component!

File: apps/web/src/pages/User/index.tsx (hunk 3 above)
Comment (JM066) [reply apps/web/src/pages/User/index.tsx:57]: Do you mean SEO for user data or project? User doesn't have many properties

File: apps/web/src/components/ProjectCard/index.tsx [hunk 4]
```diff
@@ -21,10 +22,10 @@ import {
 } from './styles';
 
 interface ProjectCardProps {
-  project: GetAllProjectsQuery['projects']['results'][0];
+  project: Partial<Project>;
 }
 
-const ProjectCard = ({ project }: ProjectCardProps) => {
+const ProjectCard = ({ project }: Partial<ProjectCardProps>) => {
```
Comment (ivanms1) [inline apps/web/src/components/ProjectCard/index.tsx:28]: don't need the partial here

**Review by ivanms1 — APPROVED** [review]
LGTM, just one small comment

File: apps/web/src/components/ProjectCard/index.tsx (hunk 4 above)
Comment (JM066) [reply apps/web/src/components/ProjectCard/index.tsx:28]: oh I thought I removed it!

**Pull request merged**